src/core/Main.pm

```diff
@@ -150,7 +150,37 @@ my sub MAIN_HELPER($retval = 0) is hidden_from_backtrace {
     @matching_candidates .=grep: {!has-unexpected-named-arguments($_.signature, $n)};
     # If there are still some candidates left, try to dispatch to MAIN
     if +@matching_candidates {
-        return $m(|@($p), |%($n));
+        # verify that we don't dispatch multiple occurences of a command line argument
+        # to a parameter that is a Scalar
```

Comment:
I'm not fully satisfied with how this PR solves the dispatch, and as you say there's still insufficient speculation on some cases.